Tests/NIOTests/ByteBufferTest.swift

```diff
@@ -1481,6 +1481,16 @@ class ByteBufferTest: XCTestCase {
         XCTAssertEqual(buf.readableBytes, viewSlice.count)
     }
 
+    func testByteBufferViewsReturnByteBuffersOfEqualLength() throws {
+        self.buf.writeString("hello")
+
+        let readableByteBuffer = self.buf.readableBytesView.getByteBuffer()
+        XCTAssertEqual(self.buf, readableByteBuffer)
+
+        let sliceByteBuffer = self.buf.viewBytes(at: 1, length: 3)?.getByteBuffer()
+        XCTAssertEqual("ell", sliceByteBuffer.flatMap { $0.getString(at: 0, length: $0.readableBytes) })
+    }
```

Comment:
can we have two more tests for the edge cases:
- the empty case
- where something matches capacity, something like this should work:

```swift
var buffer = self.allocator.buffer(16)
let capacity = buffer.capacity
buffer.writeString(String(repeating: "x", count: capacity))
XCTAssertEqual(capacity, buffer.capacity)
XCTAssertEqual(buffer.viewReadableBytes().getByteBuffer(), buffer)
```